### ISSUE TITLE
Exception NoSuchElement in Subscription.headOption()

### ISSUE THREAD

**Issue by duchoang** [mbannour/zio-mongo#9]
This line could introduce this exception when no data found for some query.
https://github.com/mbannour/zio-mongoDB/blob/08794a7a585856829bbba9707bc9bbf1986f17d0/zio-core/src/main/scala/io/github/mbannour/subscriptions/Subscription.scala#L29
Because Option constructor doesn't catch any exception raised in `next()`. I think some fix could be:
```
def headOption[F[_]]: IO[Throwable, Option[T]] = fetch.map(r => Try(r.next()).toOption)
```
or
```
def headOption[F[_]]: IO[Throwable, Option[T]] = fetch.map(r => {
  if (r.hasNext) Some(r.next()) else None
})
```

**Comment by mbannour** [comment]
Hi @duchoang ,
Thank you for raising this issue and proposing a safe ways to handle that.
I would say it good to go with the second solution :
`def headOption[F[_]]: IO[Throwable, Option[T]] = fetch.map(r => {
  if (r.hasNext) Some(r.next()) else None
})`

I will fix that.

**Comment by mbannour** [comment]
The fix is done this commit [d469d92](https://github.com/mbannour/zio-mongoDB/commit/d469d92d6d57d2a50c16a1c188459ba387ee8898)
